modules/auth/proc-adding-gitlab-oauth-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,61 @@
+
+*Name*:: Enter your application name, such as `{product-short}`.
+*Redirect URI*:: Enter the backend authentication URI set in {product-short}:
+For local development::: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+For a non-local deployment::: `pass:c,a,q[http://_<APP_FQDN>_/api/auth/gitlab/handler/frame]`
++
+NOTE: Due to a peculiarity with GitLab OAuth, ensure the URL has no trailing `/` after 'frame'.
+*Scopes*:: Select the following scopes from the list, and validate:
++
+// NOTE: including here verbatim copy from Gitlab screen.
+`read_user`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`read_repository`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`write_repository`::: Grants read/write access to repositories on private projects using Git-over-HTTP (not using the API).
+`openid`::: Grants permission to authenticate with GitLab using OpenID Connect.
+Also gives read-only access to the user's profile and group memberships.
+`profile`::: Grants read-only access to the user's profile data using OpenID Connect.
+`email`::: Grants read-only access to the user's primary email address using OpenID Connect.
++
+Save the *Application ID* and *Secret* for the next step.
+
+. Set up the Gitlab authentication provider in {product-short} configuration (`app-config.yaml`).
++
+Add the `gitlab` provider configuration to your `app-config.yaml` under the root `auth` configuration, and optionally, enable the `gitlab` provider for sign-in:
++
+.`app-config.yaml` fragment
+[source,yaml,subs="+quotes,+attributes"]
+----
+auth:
+  environment: development
+  providers:
+    gitlab:
+      development:
+        clientId: ${AUTH_GITLAB_CLIENT_ID} # <1>
+        clientSecret: ${AUTH_GITLAB_CLIENT_SECRET} # <2>
+        # audience: https://gitlab.company.com # <3>
+        # callbackUrl: https://_<APP_FQDN>_/api/auth/gitlab/handler/frame # <4>
+# signInPage: gitlab # <5>
+----
++
+<1> Enter the *Application ID* that you generated on GitLab, such as
+`4928c033ab3d592845c044a653bc20583baf84f2e67b954c6fdb32a532ab76c9`.
+<2> Enter the *Secret* that you generated on Gitlab, such as `gloas-f2c9c350759cc08346fbf94a476ae83c579c76dd629fc5eeef9dc21eedfe0475`.
+<3> Optionally, when using a self-hosted Gitlab: uncomment, and enter your GitLab instance base URL, such as
+`pass:[https://gitlab.company.com]`.
+<4> Optionally, when using a custom redirect URI: uncomment, and enter the URL matching the *Redirect URI* registered when creating your GitLab OAuth App, such as:
+* For local development: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
```

Comment:
@kadel Does it make sense to mention local development at all? I am deploying locally on OpenShift Local, and the URL is https://backstage-developer-hub-rhdh-instance.apps-crc.testing/settings

Reply:
https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/78/files#r1540657644